Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -508,36 +510,29 @@ public Solution GetOverflowSolution(EntityUid uid, PuddleComponent? puddle = nul
         if (solution.Volume == 0)
             return false;
 
-        var targets = new List<EntityUid>();
-        var reactive = new HashSet<Entity<ReactiveComponent>>();
-        _lookup.GetEntitiesInRange(coordinates, 1.0f, reactive);
-
-        // Get reactive entities nearby--if there are some, it'll spill a bit on them instead.
-        foreach (var ent in reactive)
+        var targets = _lookup.GetEntitiesInRange(coordinates, 1.0f, LookupFlags.Uncontained);
+        foreach (var ent in targets)
         {
-            // sorry! no overload for returning uid, so .owner must be used
-            var owner = ent.Owner;
-
-            // between 5 and 30%
-            var splitAmount = solution.Volume * _random.NextFloat(0.05f, 0.30f);
-            var splitSolution = solution.SplitSolution(splitAmount);
-
+            SpillSolutionOnTarget(solution, ent, _random.NextFloat(0.05f, 0.30f));
```

Comment:
Since we already know the number of targets it should be pretty easy to set the spill fraction to something like `0.5 / count--` which should evenly distribute the reagents among the targets.

Reply:
Also as above should be using fixedpoint2 instead.